upstash_vector/utils.py

```diff
@@ -14,42 +14,83 @@ def convert_to_list(obj):
     )
 
 
-def _tuple_to_vector(vector) -> Vector:
-    if len(vector) < 2 or len(vector) > 3:
-        raise ClientError("Tuple must be in the format (id, vector, metadata)")
+def _get_payload_element(
+        id: Union[int, str],
+        payload: Union[str, List[float]],
+        metadata: Dict[str, Any] = None
+) -> Union[Vector, Data]:
 
-    if len(vector) == 2:
-        return Vector(id=vector[0], vector=convert_to_list(vector[1]))
+    if isinstance(payload, str):
+        return Data(id=id, data=payload, metadata=metadata)
 
-    return Vector(id=vector[0], vector=convert_to_list(vector[1]), metadata=vector[2])
+    return Vector(id=id, vector=convert_to_list(payload), metadata=metadata)
 
 
-def _dict_to_vector(vector) -> Vector:
-    if vector["id"] is None or vector["vector"] is None:
-        raise ClientError("Vector dict must have 'id' and 'vector' fields defined.")
+def _get_payload_element_from_dict(
+        id: Union[int, str],
+        vector: List[float] = None,
+        data: str = None,
+        metadata: Dict[str, Any] = None
+) -> Union[Vector, Data]:
+    is_vector, is_data = not vector is None, not data is None
```

Comment:
```suggestion
    is_vector, is_data = vector is not None, data is not None
```
minor: for better readibility

Reply:
removed these variables because typing was getting confused below, used `is not` format in the new version